### DISCUSSION TITLE
Period priors for trigonometric functions

### DISCUSSION THREAD

**Discussion by mattmoo** [lindeloev/mcp#107] [Q&A]
Firstly, thank you for all your work on this package!

I have some time series data that seems to have an annual seasonal effect. I'm not getting a particularly satisfactory fit, and wonder whether it might be possible to set a prior for the period?

**Comment by lindeloev** [comment]
Yes, I did consider adding this. Right now, `mcp` just follows `glm` standard that only the amplitude is modelled, i.e., `parameter * sin(x)`. But technically, it's easy to do expand the model to `amplitude * sin((phase + x)/period)` with appropriate priors for the two new parameters. 

I lean towards doing this by default. Here's the dilemma: this would perhaps surprise some users because this would be nonstandard behavior in R. But the only cases I can think of where you *wouldn't* want to do this is when you know a priori that phase == 0 & period == 1, which seems rare. If you wanted this behavior anyway, you could just set `prior = list(sinx_phase_1 = 0, sinx_period_1 = 1)`. `brms` has a non-linear "module" where you explicitly write the formula *with* parameters `y ~ b_amp * sin((b_phase + x) / b_period`, but I think this is beyond the scope of `mcp`.

Right now, with mcp v0.3, you can do a very rough fix by forcing a period of 1: simply do `x = period + x / phase` with you dataset and then back-transform all regressors after fitting. Here, you infer in some other way.

You can also use `mcp` to generate the JAGS code `fit$jags_code` and then add phase/period and sample it directly using JAGS/rjags. But then you manually have to do check fits/predictions/plots, etc.

**Reply by lindeloev** [reply]
Just posted a [twitter poll](https://twitter.com/jonaslindeloev/status/1369615998500880388) :-)

**Reply by mattmoo** [reply]
Interesting, I think that I can eyeball this particular seasonal effect, but I can see that not being the case all the time.
